inc/blocks/namespace.php

```diff
@@ -73,11 +73,18 @@ function on_save_post( int $post_ID, WP_Post $post, bool $update ) : void {
 		return;
 	}
 
-	if ( $post->post_type === POST_TYPE ) {
+	$post_types = array(
+		POST_TYPE,
+		'revision',
+		'customize_changeset'
+	);
+
+	if ( in_array( $post->post_type, $post_types ) ) {
 		return;
 	}
 
-	if ( $post->post_type === 'revision' ) {
+	$post_type_object = get_post_type_object( $post->post_type );
+	if( !$post_type_object->public || !$post_type_object->show_in_rest ){
```

Comment:
We want to process the post if _either_ `public` or `show_in_rest` are true. Currently this is saying:

> if `public` is false then return early, or if `show_in_rest` is false then return early

Whereas what we really need to say is

> if `public` is and false and `show_in_rest` is also false then return early

Because we're returning early the logic is reversed - it's a bit to get your head around at first when using early returns. If the code we wanted to execute was inside the `if` block then the `||` operator would make sense.

Remember to keep an eye on spacing and code style too.

```suggestion
	if( ! $post_type_object->public && ! $post_type_object->show_in_rest ) {
```